### PR TITLE
Mineral machine revamps, constructable/modifyable unloaders.

### PR DESCRIPTION
**Pull request by ghost**
[content][tweak]
### Main accomplishments of this PR:
- A new base type of mineral machine, covering a lot of common functions such as processing or multitool menus and cutting down a lot of (but not all) copypasted code.
- Revamp of ore unloaders, to be buildable from circuits imprinted at R&D, parts that are upgradable, have in_dirs and out_dirs just like the other machines instead of some weird object searching system, and to conveyor_act() with things on the other side. (including other unloaders, chaining them, although this is ~~not~~ recommended in most cases)
- Compressing the stacking/smelting multitool menus down to one in the base item type for setting directions, now allowing it on unloaders.

These have been somewhat tested, unloaders can be built from their circuits, rnd design datums usually show up, unloaders, smelters and stacking machines work perfectly.

### Stuff that needs to be done:
- [x] ~~Removing all of those old in and out position items from maps.~~
- [x] ~~Changing all in_dirs and out_dirs of each unloader. (and any R&D machines that used em)~~
- [x] ~~Deleting position items from the code altogether.~~
- Allowing unloaders to filter types. (in this PR or another one)

:cl:
 * rscadd: Mineral unloading machines are now buildable from circuit imprinters.
 * rscadd: Mineral unloading machines now have 3 matter bins and a capacitor, all upgradable for faster unloading and lower power usage.
 * rscadd: Mineral unloading machines can now have their directions tweaked with a multitool with ease.
 * rscadd: Mineral unloading machines work like conveyors on machines on the other end.
 * tweak: New common mineral machine item, cutting down a lot of redundant code and variables.

### CODE REVIEW

**Comment by Eneocho** [comment]
Obligatory Prepare for Unforeseen Consequences post.

**Comment by ghost** [comment]
I have no idea why icon diffs are failing, I didn't even touch icons

**Comment by ghost** [comment]
Oh right it was just a CI issue

File: code/modules/research/designs/boards/machine_misc.dm [hunk 1]
```diff
@@ -87,6 +87,16 @@
 	category = "Machine Boards"
 	build_path = /obj/item/weapon/circuitboard/stacking_unit
 
+/datum/design/stacking_unit
```
Comment (Kurfursten) [inline code/modules/research/designs/boards/machine_misc.dm]: whoopsie

Comment (ghost) [reply code/modules/research/designs/boards/machine_misc.dm]: Fixed

**Comment by Eneocho** [comment]
Perhaps make it so upgrading the smelter/recycler gets you a bit extra material per processed ore. So a normal machine gets you 1 metal, but with max level parts you get 1.5 metal per ore, so there's an incentive to upgrade the machinery. It'd be explained IC as the refining process getting more efficient so there's less wasted material.

**Pull request merged**